### PR TITLE
Cleanup ClientSession builder

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: client/trino-client/src/main/java/io/trino/client/ClientSession.java [hunk 1]
```diff
@@ -100,8 +100,8 @@ private ClientSession(
             Optional<String> encoding)
     {
         this.server = requireNonNull(server, "server is null");
-        this.principal = requireNonNull(principal, "principal is null");
         this.user = requireNonNull(user, "user is null");
+        this.sessionUser = requireNonNull(sessionUser, "user is null");
```
Comment (ebyhr) [inline client/trino-client/src/main/java/io/trino/client/ClientSession.java:104]:
```suggestion
        this.sessionUser = requireNonNull(sessionUser, "sessionUser is null");
```

File: client/trino-client/src/main/java/io/trino/client/ClientSession.java [hunk 2]
```diff
@@ -356,9 +356,9 @@ public Builder server(URI server)
             return this;
         }
 
-        public Builder user(Optional<String> user)
+        public Builder sessionUser(Optional<String> user)
```
Comment (ebyhr) [inline client/trino-client/src/main/java/io/trino/client/ClientSession.java:359]:
```suggestion
        public Builder sessionUser(Optional<String> sessionUser)
```

File: client/trino-client/src/main/java/io/trino/client/ClientSession.java (hunk 1 above)
Comment (wendigo) [reply client/trino-client/src/main/java/io/trino/client/ClientSession.java:104]: Good catch!

**Review by losipiuk — APPROVED** [review]

File: client/trino-client/src/main/java/io/trino/client/ClientSession.java [hunk 3]
```diff
@@ -368,9 +368,9 @@ public Builder authorizationUser(Optional<String> authorizationUser)
             return this;
         }
 
-        public Builder principal(Optional<String> principal)
+        public Builder user(Optional<String> user)
```
Comment (ebyhr) [inline client/trino-client/src/main/java/io/trino/client/ClientSession.java:371]: nit: The current order bewteen fields and setters differs. I prefer using the same order.
```java
private Optional<String> user = Optional.empty();
private Optional<String> sessionUser = Optional.empty();
private Optional<String> authorizationUser = Optional.empty();
...
public Builder sessionUser(Optional<String> sessionUser)
public Builder authorizationUser(Optional<String> authorizationUser)
public Builder user(Optional<String> user)
```
Please feel free to ignore as pre-existing.

Comment (wendigo) [reply client/trino-client/src/main/java/io/trino/client/ClientSession.java:371]: I'll apply this change once the CI is done, thanks!

**Pull request merged**